fluster/cluster.py

```diff
@@ -38,7 +42,50 @@ def __init__(self,
                                       multiplier=penalty_box_wait_multiplier)
         self.active_clients = self._prep_clients(clients)
         self.initial_clients = {c.pool_id: c for c in clients}
+        self.clients = cycle(self.initial_clients.values())
         self._sort_clients()
+        # maintenance work is done each "tick" when iterating
+        self._tick_interval = 2 * len(clients)
+        self._ticks = 0
+
+    def __iter__(self):
+        """Updates active clients each time it's iterated through."""
+        self._prune_penalty_box()
+        return self
+
+    def __next__(self):
+        """Always returns a client, or raises an Exception if none are available."""
+        # raise Exception if no clients are available
+        if len(self.active_clients) == 0:
+            raise ClusterEmptyError('All clients are down.')
+
+        # always return something
+        nxt = None
+        while nxt is None:
+            nxt = self._next_helper()
```

Comment:
Doesn't necessarily need to be a separate function since it's only called once. Also the check for an empty cluster should probably be part of this loop so that it's impossible for this to become an infinite loop.